### DISCUSSION TITLE
Reusable component className props Custom Styling didn't working in react

### DISCUSSION THREAD

**Discussion by febriaricandra** [tailwindlabs/tailwindcss#8856] [Help] [answered]
hey i just configured my custom colors on tailwind.config but when i try to pass it as classname props styling my custom colors didn't display on the browser. need help!
![image](https://user-images.githubusercontent.com/52602319/178923486-793f8ccc-27ba-457b-9291-3db11a5933e4.png)

**Comment by arisac** [comment]
That should work. Try dig into the `Styles` tab in the inspector to see if it get applied or override by other styles.
<img width="400" alt="image" src="https://user-images.githubusercontent.com/63835273/178927266-d4573ac9-26ef-4ce9-8208-3697b316f6a8.png">

**Reply by febriaricandra** [reply]
this is my code btw, is it good in tailwind? i am trying to dynamic styling it in react js and then when i run the custom color didn't displayed in the browser
![image](https://user-images.githubusercontent.com/52602319/178927819-524ee540-53a7-4a6b-964f-d5840af9c045.png)
![image](https://user-images.githubusercontent.com/52602319/178927891-eec2e2d4-67af-436d-b8d1-27b369d28ceb.png)

**Reply by arisac** [reply]
That will not work since TailwindCSS is actually build the css properties into the css file so it can be used. It is not js on the client side.
During the build, TailwindCSS doesn't recognize `text-${textColor}` so the following won't be built into the CSS file.
```
.text-primary {
    --tw-text-opacity: 1;
    color: rgb(56 101 244 / var(--tw-text-opacity));
}
```

A workaround for this is actually using `className="text-primary"` somewhere else so it can be created into the CSS file.

However, I'm not recommending that. Consider to build a better BTN component.

**Reply by febriaricandra** [reply]
is there another way to create reusable component with tailwind?

**Reply by arisac** [reply]
You can pass the classnames to the child components.
```jsx example
export function PrimaryButton({ className = '', children }) {
  return <button className={`px-3 py-2 text-2xl uppercase ${className}`}>{children}</button>
}

export function SecondaryButton({ className = '', children }) {
  return <button className={`px-2 py-1 text-lg ${className}`}>{children}</button>
}
```

**Reply by febriaricandra** [reply]
that's work well, and it's like paradox styling lmao! thank you bro

**Reply by arisac** [reply]
it's just a hack for your specific use case.
however, I don't recommend building a button component like this.
it's not less code than using just `<button />`, and not really a *Systematic UI*

**Reply by febriaricandra** [reply]
i got it bro! do you have any good references regarding that? systematic UI or some advices for building other components.

**Reply by arisac** [reply]
> i got it bro! do you have any good references regarding that? systematic UI or some advices for building other components.

This question is no longer part of TailwindCSS issue. *Systematic UI* is related to UX/UI design. 
For making react UI components, you can always google the most popular *react UI frameworks* and learn from them.

**Reply by zslabs** [reply]
Instead of passing a custom classname, just make the props output the entire classname that customized the colors. Tailwind can't parse template literals. This way your styles all remain in the same spot, with less of a chance of conflicts from custom classes passed in elsewhere.

**Comment by arisac** [comment, marked as answer]
@febriaricandra please close this discussion and mark as answered so other people don't need come in to see if you still help.

**Reply by febriaricandra** [reply]
sorry bro, i forgot to mark this issues
